### ISSUE TITLE
Creating points in 3D on a 2D layer

### ISSUE THREAD

**Issue by brisvag** [napari/napari#3719]
## 🐛 Bug

#3508 Introduced the ability to create points in 3D. However, if one tries to add points while in 3D to a 2D layer, the behaviour is quite surprising.

https://user-images.githubusercontent.com/23482191/144088048-769f05a3-a754-4bc8-890e-a02761fb642c.mp4

## To Reproduce

```python
import napari
v = napari.Viewer(ndisplay=3)
pl = v.add_points()
pl.mode = 'add'
# click around
```

## Expected behavior

Either we should upgrade the layer to 3D (my preferred behaviour, but I understand this might be disruptive to some) or we should match dimensions correctly (currently moving up and down results in points being positioned more left or right).

**Comment by kevinyamauchi** [comment]
Ah yeah, that's definitely not ideal...I am actually not totally sure how a 2D layer is displayed in 3D. Does it align the 3D layers 0 and 1 axis with the 0 and 1 axis of the 3d layer?

I think in any case it will be a little confusing, as it won't be clear where the plane of the 2D points player is. Perhaps it's best to disallow adding points to 2D layers in 3D?

**Comment by brisvag** [comment]
The problem actually only shows up if no 3D layer is present.

```python
import napari
import numpy as np

v = napari.Viewer(ndisplay=3)
il = v.add_image(np.random.rand(10, 10, 10))
pl = v.add_points()
pl.mode = 'add'
v.axes.visible = True
```

This works fine. If however you delete the image layer, things start misbehaving again. It seems that axes alignment is doing something weird!

**Comment by brisvag** [comment]
#4096 was actuallly a duplicate of this, and was fixed (or at least, "guarded") by #4184.
